### DISCUSSION TITLE
Estimate price elasticities for product categories

### DISCUSSION THREAD

**Discussion by IskanderNakipov** [DoubleML/doubleml-for-py#135] [Q&A] [answered]
Hi!

I'm trying to use DoubleML to estimate price elasticity on retail data. I'm using PLR in the following form:

**log(Sales) = elasticity * log(price) + g(x)**

But this model only provides one elasticity for all products. I would like, however, to estimate elasticity for each product category.

How should I do that? My only idea is to fit estimator separately on data for each category, but I would like to use all the data available to train nuisance estimators to achieve better quality

**Comment by MalteKurz** [comment, marked as answer]
Thanks for your interest in `DoubleML`.

You may want to take a look into "Estimation and Inference on Heterogeneous Treatment Effects in High-Dimensional Dynamic Panels" https://arxiv.org/abs/1712.09988. The application in Section 6 also considers some kind of groups via interacting indicators for the product category with the prices. While `DoubleML` currently does not come with an explicit implementation for panel data, we support multiple treatment variables.

More general GATEs (group average treatment effects) could be of interest for your use case. This can also be estimated with double machine learning, see for example "Debiased Machine Learning of Conditional Average Treatment Effects and Other Causal Functions" (https://arxiv.org/abs/1702.06240). We have some more convenient API for GATEs estimation in our pipeline but the implementation is not yet finalized. I will let you know when it's added to the package.

**Reply by IskanderNakipov** [reply]
Thanks a lot, seems like GATEs are exactly what I need! Looking forward to see them in DoubleML :)
